homeassistant/config_entries.py

```diff
@@ -2726,6 +2726,39 @@ def is_matching(self, other_flow: Self) -> bool:
         """Return True if other_flow is matching this flow."""
         raise NotImplementedError
 
+    @property
+    def _reauth_entry_id(self) -> str:
+        """Return reauth entry id."""
+        if self.source != SOURCE_REAUTH:
+            raise ValueError(f"Source is {self.source}, expected {SOURCE_REAUTH}")
+        return self.context["entry_id"]  # type: ignore[no-any-return]
+
+    @callback
+    def _get_reauth_entry(self) -> ConfigEntry:
+        """Return the reauth config entry linked to the current context."""
+        if self.source != SOURCE_REAUTH:
+            raise ValueError(f"Source is {self.source}, expected {SOURCE_REAUTH}")
```

Comment:
We call `_reauth_entry_id` which already has this check and raises the same error, why do we check it here too?